lib/src/thresholdruledevice.h

```diff
@@ -100,22 +106,120 @@ class ThresholdRuleDevice : public Rule
         return 0;
     }
 
-    int evaluate(const MetricList& /* metricList */, PureAlert& /* pureAlert */)
+    int evaluate(const MetricList& metricList, PureAlert& pureAlert)
     {
-        // INTENTIONALLY We do not evaluate this rule at all
-        // It is evaluated in NUT-agent somewhere
-        // rules are here, just to provide webUI access to the rule representation
+        // ASSUMPTION: constants are in values
+        //  high_critical
+        //  high_warning
+        //  low_warning
+        //  low_critical
+
+        log_debug("ThresholdRuleSimple::evaluate %s", _name.c_str());
+
+    #if 0 //DBG, trace _outcomes
+        log_debug("%s: outcomes (size: %zu)", _name.c_str(), _outcomes.size());
+        for (auto& outcome : _outcomes) {
+            log_debug("%s: %s", outcome.first.c_str(), outcome.second.str().c_str());
+        }
+    #endif
+
+        const auto GV = getGlobalVariables();
+        const MetricInfo lastMetric = metricList.getLastMetric();
```

Comment:
my opinion: too many copy/paste codes here

Reply:
I agree, I made a lambda at the beginning but not easier with the test inferior or superior (complicate too much the code)